### PR TITLE
ECO-3656: Computop PayPal Express.

### PR DESCRIPTION
**Pull request by kkicheglovspryker**
- Developer(s): @kkicheglovspryker

- Ticket: https://spryker.atlassian.net/browse/ECO-3656

- Release Group: https://release.spryker.com/release-groups/view/3540

- PR Overview: https://release.spryker.com/release/pull-request-eco/ComputopApi/9

- merge: squash

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   ComputopApi            | minor                 |                       |

-----------------------------------------

##### Change log

Improvements

- Introduced `ComputopApiFacade::performPayPalExpressPrepareApiCall()` to perform `PayPalExpressPrepareRequest` to Computop API and modify `QuoteTransfer`.
- Introduced `ComputopApiFacade::performPayPalExpressCompleteApiCall()` to perform `PayPalExpressCompleteRequest` to Computop API and modify `QuoteTransfer`.
- Introduced `ComputopApiClient::sendPayPalExpressPrepareRequest()`.
- Introduced `ComputopApiClient::sendPayPalExpressCompleteRequest()`.
- Introduced `ComputopApiPayPalExpressPrepareResponse` transfer object to hold Computop API response for `PayPalExpressPrepareRequest`.
- Introduced `ComputopApiPayPalExpressComplete` transfer object to hold Computop API request for `PayPalExpressCompleteRequest`.
- Introduced `ComputopPayPalExpressPayment` transfer object.
- Introduced `ComputopApiPayPalExpressCompleteResponse` transfer object to hold Computop API response for `PayPalExpressCompleteRequest`.
- Introduced `ComputopApiConfig::getPayPalExpressPrepareActionUrl()` to get PayPal Express prepare action URL.
- Introduced `ComputopApiConfig::getPayPalExpressCompleteActionUrl()` to get PayPal Express complete action URL.
- Added `ZedRequestClientInterface` to dependencies.
- Added `UtilTextServiceInterface` to dependencies.

### CODE REVIEW

File: src/SprykerEco/Service/ComputopApi/Dependency/Service/ComputopApiToUtilTextServiceBridge.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+
+namespace SprykerEco\Service\ComputopApi\Dependency\Service;
+
+class ComputopApiToUtilTextServiceBridge implements ComputopApiToUtilTextServiceInterface
+{
+    /**
+     * @var \Spryker\Service\UtilText\UtilTextServiceInterface
+     */
+    protected $textService;
```
Comment (DmitryLymarenko) [inline src/SprykerEco/Service/ComputopApi/Dependency/Service/ComputopApiToUtilTextServiceBridge.php:15]:
```suggestion
    protected $utilTextService;
```

File: src/SprykerEco/Zed/ComputopApi/Business/Request/ExpressCheckout/AbstractPayPalExpressRequest.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    /**
+     * @var \SprykerEco\Zed\ComputopApi\Business\Converter\ConverterInterface
+     */
+    protected $converter;
+
+    /**
+     * @var \SprykerEco\Zed\ComputopApi\Business\Mapper\ExpressCheckout\PayPalExpressMapperInterface
+     */
+    protected $payPalExpressMapper;
```
Comment (DmitryLymarenko) [inline src/SprykerEco/Zed/ComputopApi/Business/Request/ExpressCheckout/AbstractPayPalExpressRequest.php:30]: same as above
looks like specific child dependency

File: tests/SprykerEcoTest/Zed/ComputopApi/_support/Business/Mapper/ExpressCheckout/PayPalExpressMapperTestHelper.php [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        $totalsTransfer = new TotalsTransfer();
+        $quoteTransfer->setTotals($totalsTransfer);
+
+        return $quoteTransfer;
+    }
+
+    /**
+     * @return \PHPUnit_Framework_MockObject_MockObject|\SprykerEco\Zed\ComputopApi\ComputopApiConfig
+     */
+    public function createComputopApiConfigMock()
```
Comment (DmitryLymarenko) [inline tests/SprykerEcoTest/Zed/ComputopApi/_support/Business/Mapper/ExpressCheckout/PayPalExpressMapperTestHelper.php:58]: please add a typehint here and below

File: src/SprykerEco/Zed/ComputopApi/Business/Request/ExpressCheckout/AbstractPayPalExpressRequest.php (hunk 2 above)
Comment (kkicheglovspryker) [reply src/SprykerEco/Zed/ComputopApi/Business/Request/ExpressCheckout/AbstractPayPalExpressRequest.php:30]: this property is used in both child classes

File: tests/SprykerEcoTest/Zed/ComputopApi/_support/Module/EnvironmentModule.php [hunk 4]
```diff
@@ -13,7 +13,7 @@
 class EnvironmentModule extends Environment
 {
     /**
-     * @inheritdoc
+     * @inheritDoc
```
[the remark this answers is not in the document]
Comment (kkicheglovspryker) [reply tests/SprykerEcoTest/Zed/ComputopApi/_support/Module/EnvironmentModule.php:16]: This phpdoc was auto-generated but ok (:

**Review by DmitryLymarenko — APPROVED** [review]

**Review by DmitryLymarenko — COMMENTED** [review]
`src/SprykerEco/Service/ComputopApi/Dependency/Service/ComputopApiToUtilTextServiceBridge.php` introduction should be mentioned in changelogs

according to https://spryker.atlassian.net/wiki/spaces/CORE/pages/523730976/Release+Notes+and+Changelogs

**Review by DmitryLymarenko — APPROVED** [review]

**Pull request merged**